presto-spi/src/main/java/com/facebook/presto/spi/PrimaryKeyConstraint.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class PrimaryKeyConstraint<T>
```

Comment:
Just curious: What's the use case for having a separate `PrimaryKeyConstraint`, that's basically the same as `UniqueConstraint`? What does primary key mean in the context of Hive?

Reply:
Should all table constraints related code belong to a sub package of spi?

Reply:
Yeah, good question. Primary and Unique key constraints are similar in that they both can be used in optimizations that involve uniqueness; however, the former can also be used in optimizations that involve referential integrity as described in the future work section of the design doc. So we do not want to lose the information that one is a primary key.